packages/@sanity/state-router/src/useRouterState.ts

```diff
@@ -0,0 +1,17 @@
+import {useContext, useEffect, useState} from 'react'
+import {RouterState} from './components/types'
+import {RouterContext} from './RouterContext'
+
+export const useRouterState = (): RouterState => {
+  const router = useContext(RouterContext)
+  const [routerState, setState] = useState<RouterState>(router.getState())
+
+  useEffect(() => {
+    // subscribe() returns an unsubscribe function, so this'll handle unmounting
+    return router.channel.subscribe(() => {
+      setState(router.getState())
+    })
+  }, [router])
```

Comment:
This should probably be `[router.channel]` to avoid excessive resubscribes (afaik `router` changes more frequently)?

Reply:
Good point, will fix 👍

Reply:
Ah, it was because it’s using `router.getState()` below there.

Reply:
Ah, I see! Probably better to push down the new state at some point, but let's leave that for later 👍🏼